### ISSUE TITLE
Keras Attention Layer not Supported When Importing to DL4J

### ISSUE THREAD

**Issue by sam-hieken** [deeplearning4j/deeplearning4j#9908]
#### Issue Description

Good afternoon,

It looks like the Attention layer in Keras isn't yet supported when importing a Keras model into DL4J. Looks like a ticket doesn't already exist for this so I figured I'd open one.

#### Version Information
* Deeplearning4j version: 1.0.0-M2.1

#### Additional Information

Code:

        public static void main(String[] args) throws IOException, UnsupportedKerasConfigurationException, InvalidKerasConfigurationException {
		String fullModel = "C:\\Users\\Sam\\Desktop\\model.h5";
		ComputationGraph model = KerasModelImport.importKerasModelAndWeights(fullModel);
		
	        System.out.println(model.output(testingData()));
	}

Exception:
`Warning: Versions of org.bytedeco:javacpp:1.5.6 and org.bytedeco:hdf5:1.12.1-1.5.7 do not match.
Exception in thread "main" org.deeplearning4j.nn.modelimport.keras.exceptions.UnsupportedKerasConfigurationException: Unsupported keras layer type Attention. Please file an issue at https://github.com/eclipse/deeplearning4j/issues.
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasLayerUtils.getKerasLayerFromConfig(KerasLayerUtils.java:337)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.prepareLayers(KerasModel.java:223)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:165)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:97)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildModel(KerasModelBuilder.java:311)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasModelAndWeights(KerasModelImport.java:167)
	at ml.TransformerKeras.main(TransformerKeras.java:16)`

I'm just using a simple model with a basic [Attention layer](https://keras.io/api/layers/attention_layers/attention/). If you'd like to see the Python code, please let me know.

**Comment by agibsonccc** [comment]
@sam-hieken  could you send me the model or specify some code to save it as well as specify the keras version? Thanks!

**Comment by sam-hieken** [comment]
@agibsonccc Of course, here's the Keras model:

	inputs = keras.Input(shape=(22,), dtype="float32", name="inputs")

	x = layers.Dense(32, activation="relu")(inputs)

	attn = layers.Attention(dropout=0.2)([x, x])
	x = layers.Concatenate()([x, attn])


	x = layers.Dense(32, activation="relu")(x)

	outputs = layers.Dense(1, activation="relu", name="outputs")(x)

	model = keras.Model(inputs=inputs, outputs=outputs)

	# Compile the model
	model.compile(loss="mean_squared_error", optimizer=(tf.keras.optimizers.Adam(learning_rate=0.005)) )

	model.save('./model.h5')

Keras Version: 2.11.0

**Comment by agibsonccc** [comment]
@sam-hieken  thanks. After looking in to this we have a few different kinds of attention implemented. I need to add 1 more mode (additive) and we should be able to import this. I'm working on this now but it'll end up being more than just "add this config" I'll be adding some new features to cover for this. Your layer itself appears to be pretty simple.

My main concern will be the testing but I think I can make this show up in the next release. JFYI.

**Comment by sam-hieken** [comment]
@agibsonccc Haha yea simple is an understatement, no rush releasing it or anything I was just messing around with importing a Keras model into DL4J. Thanks

**Comment by agibsonccc** [comment]
Closing — this has been addressed in the upcoming release (branch ag_new_release_updates_2). The relevant code has been rewritten/fixed.
